corona-warn-app/cwa-app-android
# [Feature] Family test list (EXPOSUREAPP-12321, EXPOSUREAPP-12327)

### Whats new

Family test list screen.

![Screenshot 2022-03-30 at 22 55 54](https://user-images.githubusercontent.com/64849422/160920405-d7aae2fe-a7e3-4635-a838-9fd88ae1e5f8.png)

### Story

- [Story](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5895)
- [Family test list UI](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12321)
- [Dismiss notification badge logic](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12327)

### How to test

1. Take a cup of tea/coffee - this would be a long ride
2. Using INT env scan both PCR and RAT types for POSITIVE, NEGATIVE, INVALID
3. Go to family tests fragment and check, that every test appeared as it should be
4. Check all tests with notification badge
5. Navigate back to main fragment
6. Navigate back again to family tests fragment - check that all notification badges are gone
7. Switch to WRU env and check tests
8. Delete one test by swiping left and confirm in alert dialog
9. Delete one test by swiping right and confirm in alert dialog
10. Delete one test by pressing delete button (should be at least one in REDEEMED state)
11. Delete all left tests by using menu in top right corner (confirmation dialog should also appear)
12. Go to recycle bin and restore tests
13. Go to phone system settings and time travel (+10 years will be just fine)
14. Go back to family tests fragment - NEGATIVE PCR should became OUTDATED with delete button
15. Press that delete button
16. Delete all tests
17. Switch to Night Mode
18. Anger
19. Denial
20. Acceptance
21. Repeat steps 2-16 and check if everything is okay in Night Mode

### To Do

- [ ] Navigation to test details
- [x] Update Invalid and Outdated card icons
- [x] Update confirmation dialog text
